src/NuGet.Indexing/VerifiedPackages.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using FrameworkLogger = Microsoft.Extensions.Logging.ILogger;
+using System.IO;
+
+namespace NuGet.Indexing
+{
+    public static class VerifiedPackages
+    {
+        public static HashSet<string> Load(string fileName, ILoader loader, FrameworkLogger logger)
+        {
+            try
+            {
+                using (var reader = loader.GetReader(fileName))
+                {
+                    return Parse(reader);
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("Unable to load {FileName} as deserialization threw: {Exception}", fileName, ex);
+
+                throw;
+            }
+        }
+
+        public static HashSet<string> Parse(JsonReader reader)
+        {
+            // The file should contain an array of strings, such as: ["Package1","Package2", ...]
+            reader.Read();
+            ThrowIfNotExpectedToken(reader, JsonToken.StartArray);
+
+            // Read all of the package ID strings from the array.
+            string packageId = reader.ReadAsString();
+            var result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            while (packageId != null)
+            {
+                result.Add(String.Intern(packageId));
```

Comment:
Interning strings is a pattern used throughout the search service. `String.Intern` is to reduce memory pressure, not to improve performance. Each reload allocates a bunch of strings (things like package IDs, versions, etc..) that already exist in memory from the previous index. By interning the string, we can reuse the memory that was allocated the first that string was encountered. The memory allocated during the index reload will then be collected on the next garbage collection. If we didn't intern the strings, the strings would be allocated on each reload and would need to be freed on the subsequent reload. Since the time between reloads is in the order of minutes, the strings would likely get promoted to the next generation of memory. Garbage collection is cheaper when memory is only used briefly.